src/ServiceControl.Config/UI/Upgrades/AddNewAuditInstanceViewModelValidator.cs

```diff
@@ -0,0 +1,68 @@
+    {
+        public AddNewAuditInstanceViewModelValidator()
+        {
+            var serviceControlInstances = InstanceFinder.ServiceControlInstances();
+            var serviceControlAuditInstances = InstanceFinder.ServiceControlAuditInstances();
+
+            RuleFor(x => x.ServiceControlAudit.ServiceAccount)
+                .NotEmpty()
+                .When(x => x.SubmitAttempted);
+
+            RuleFor(x => x.ServiceControlAudit.PortNumber)
+                .NotEmpty()
+                .ValidPort()
+                .PortAvailable()
+                .MustNotBeIn(x => serviceControlInstances.UsedPorts(x.ServiceControlAudit.InstanceName))
+                .MustNotBeIn(x => serviceControlAuditInstances.UsedPorts(x.ServiceControlAudit.InstanceName))
+                .NotEqual(x => x.ServiceControlAudit.DatabaseMaintenancePortNumber)
+                .WithMessage(Validations.MSG_MUST_BE_UNIQUE, "Ports")
+                .When(x => x.SubmitAttempted);
+
+            RuleFor(x => x.ServiceControlAudit.DatabaseMaintenancePortNumber)
+                .NotEmpty()
+                .ValidPort()
+                .PortAvailable()
+                .MustNotBeIn(x => serviceControlInstances.UsedPorts(x.ServiceControlAudit.InstanceName))
+                .MustNotBeIn(x => serviceControlAuditInstances.UsedPorts(x.ServiceControlAudit.InstanceName))
+                .NotEqual(x => x.ServiceControlAudit.PortNumber)
+                .WithMessage(Validations.MSG_MUST_BE_UNIQUE, "Ports")
+                .When(x => x.SubmitAttempted);
+
+            RuleFor(x => x.ServiceControlAudit.DestinationPath)
+                .NotEmpty()
+                .ValidPath()
+                .MustNotBeIn(x => serviceControlInstances.UsedPaths(x.ServiceControlAudit.InstanceName))
+                .MustNotBeIn(x => serviceControlAuditInstances.UsedPaths(x.ServiceControlAudit.InstanceName))
+                .WithMessage(Validations.MSG_MUST_BE_UNIQUE, "Paths")
+                .When(x => x.SubmitAttempted);
+
+            RuleFor(x => x.ServiceControlAudit.DatabasePath)
+                .NotEmpty()
+                .ValidPath()
+                .MustNotBeIn(x => serviceControlInstances.UsedPaths(x.ServiceControlAudit.InstanceName))
+                .MustNotBeIn(x => serviceControlAuditInstances.UsedPaths(x.ServiceControlAudit.InstanceName))
+                .WithMessage(Validations.MSG_MUST_BE_UNIQUE, "Paths")
+                .When(x => x.SubmitAttempted);
+
+            RuleFor(x => x.ServiceControlAudit.AuditForwarding)
+                .NotNull().WithMessage(Validations.MSG_SELECTAUDITFORWARDING);
+
+            RuleFor(x => x.ServiceControlAudit.AuditQueueName)
+                .NotEmpty()
+                .NotEqual(x => x.ServiceControlAudit.AuditForwardingQueueName).WithMessage(Validations.MSG_UNIQUEQUEUENAME, "Audit Forwarding")
+                .When(x => x.SubmitAttempted && x.ServiceControlAudit.AuditQueueName != "!disable");
```

Comment:
We wouldn't have an Audit Queue `!disable` in an Audit Instance. You'd just remove the audit instance

Reply:
I've copied the validator code verbatim from the existing validation logic of the Add screen. Should it be removed altogether or just here?